akka-http-core/src/main/scala/akka/http/impl/settings/ConnectionPoolSettingsImpl.scala

```diff
@@ -76,10 +72,6 @@ private[akka] object ConnectionPoolSettingsImpl extends SettingsCompanionImpl[Co
       c.getFiniteDuration("max-connection-backoff"),
       c.getPotentiallyInfiniteDuration("idle-timeout"),
       ClientConnectionSettingsImpl.fromSubConfig(root, c.getConfig("client")),
-      c.getString("pool-implementation").toLowerCase match {
```

Comment:
We could keep a warning if the setting is still found to be set to `legacy`. We have no logger here, though, so it didn't seem worth the effort.